packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -81,13 +103,41 @@ export interface ExtendedDataSourceProps {
 }
 
 /**
- * Abstract AppSync datasource implementation. Do not use directly but use subclasses for concrete datasources
+ * An Interface for Data Sources
  */
-export abstract class BaseDataSource extends Construct {
+export interface IDataSource extends IConstruct {
+  /**
+   * the name of the data source
+   */
+  readonly name: string;
+  /**
+   * the arn of the data source
+   */
+  readonly arn: string;
```

Comment:
are any of these attributes of the CloudFormation resource by any chance?
if so, they should be annotated with `@attribute` - check out our repo for plenty of prior art there

Reply:
they are but i didn't make the BaseDataSource a Resource... we could but it would introduce some issues..